testing/test_main.py

```diff
@@ -75,3 +77,10 @@ def pytest_sessionfinish():
         assert result.ret == ExitCode.NO_TESTS_COLLECTED
     assert result.stdout.lines[-1] == "collected 0 items"
     assert result.stderr.lines == ["Exit: exit_pytest_sessionfinish"]
+
+
+@pytest.mark.parametrize("basetemp", ["", "."])
```

Comment:
We should also add a parent directory here for testing, and probably better to change the directory to a temporary directory to ensure we have more control over cwd, something like:

```python
@pytest.mark.parametrize("basetemp", ["", ".", ".."])
def test_validate_basetemp_fails(tmp_path, basetemp, monkeypatch):
    monkeypatch.chdir(str(tmp_path))
    msg = "basetemp should not be '' or . or any parent folder of the cwd"    
    with pytest.raises(argparse.ArgumentTypeError, match=msg):
        if basetemp:
            p = tmp_path / basetemp
        else:
            p = basetemp
        validate_basetemp(p)


@pytest.mark.parametrize("basetemp", ["foo", "foo/bar"])
def test_validate_basetemp_ok(tmp_path, basetemp, monkeypatch):
    monkeypatch.chdir(str(tmp_path))
    validate_basetemp(tmp_path / basetemp)        
```

Also an integration test would be nice:

```python
def test_validate_basetemp_integration(testdir):
    result = testdir.runpytest("--basetemp=.")    
    result.stderr.fnmatch_lines("*basetemp should not be*")
```

Reply:
done. updated tests.